### PR TITLE
Allow setup custom exception handlers for failing jobs

### PR DESCRIPTION
**Pull request by nbulaj**
Allow to setup custom exception handlers in case anybody needs more actions and/or logging data when processor fails.

Idea with multiple handlers [taken from the Sidekiq ](https://github.com/sidekiq/sidekiq/blob/main/lib/sidekiq/config.rb#L268)

Fixes also to https://github.com/ruby-shoryuken/shoryuken/issues/544 & #308

### CODE REVIEW

**Comment by phstc** [comment]
Solid! Thanks for the contribution.

File: lib/shoryuken/processor.rb [hunk 1]
```diff
@@ -22,8 +22,7 @@ def process
         end
       end
     rescue Exception => ex
-      logger.error { "Processor failed: #{ex.message}" }
-      logger.error { ex.backtrace.join("\n") } unless ex.backtrace.nil?
+      Array(Shoryuken.exception_handlers).each { |handler| handler.call(ex, queue, sqs_msg) }
```
Comment (phstc) [inline lib/shoryuken/processor.rb:25]: Is there a way you can test this too? Maybe here https://github.com/ruby-shoryuken/shoryuken/blob/main/spec/shoryuken/processor_spec.rb

Comment (nbulaj) [reply lib/shoryuken/processor.rb:25]: This actually is tested in `spec/shoryuken/default_exception_handler_spec.rb` with InlineExecutor, but I can add more tests

Comment (phstc) [reply lib/shoryuken/processor.rb:25]: nvm I misread that. It looks great! Thank you again.

**Pull request merged**